solr/solr-ref-guide/src/nested-documents.adoc

```diff
@@ -264,11 +273,12 @@ For the upcoming examples, assume the following documents have been indexed:
   }
 ----
 
-==== Combining Block Join Query Parser with Child Doc Transformer
-  * The combination of these two features enable seamless creation of powerful queries. +
-  For example, querying posts which are under a page tagged as a job, contain the words "Search Engineer".
-  The comments for matching posts can also be fetched, all done in a single Solr Query.
-  `q=+{!child of='-\_nest_path_:* *:*'}+tags:"jobs" &fl=*,[child]
+==== Combining Block Join Query Parsers with Child Doc Transformer
+The combination of these two features enable seamless creation of powerful queries. +
+For example, querying posts which are under a page tagged as a job, contain the words "Search Engineer".
+The comments for matching posts can also be fetched, all done in a single Solr Query.
+
+ * `q=+{!child of='-\_nest_path_:* *:*'}+tags:"jobs" &fl=*,[child]
```

Comment:
did you test (manually) that?  That query looks dubious... why the \*:\* in there.  I get the other part would probably work (and I think I wrote that in another JIRA issue recently)

Reply:
Yeah,
I did test it manually and observed odd results.
When using +{!child of='-\_nest_path_:*'}+tags:"jobs" no documents are found.
This seems to be a problem with the way queries are optimized.
When there is only a NOT clause in the "of" filter, the query is optimized to a MatchNoDocsQuery.

Reply:
oh right.  So this is a small bug / oversight in the block join query parser; it ought to call org.apache.solr.search.QueryUtils#makeQueryable .  CC @mkhludnev